### ISSUE TITLE
Extend fix for years less than 1678 for non-"days since" time units

### ISSUE THREAD

**Issue by spencerkclark** [spencerahill/aospy#71]
Currently our workaround for dates with years less than 1678 makes the assumption that the original units of the time dimension in the read-in file were `"days since ..."`.  As part of the workaround we hard-code the units to be `"days since 1900-01-01 00:00:00"` (see https://github.com/spencerahill/aospy/blob/develop/aospy/calc.py#L417-L418) and then decode the CF properties.  This causes problems for high-frequency output, because while the units of the time dimension in the original file might be `"hours since ..."`, they get changed to `"days since 1900-01-01 00:00:00"`.  This means that xarray interprets the time interval between outputs as days (rather than hours in this example); e.g. 2920 3-hourly outputs get interpreted to be 24 years (rather than one year).  
- A straightforward fix would be to allow the user to specify another argument when setting up a calc to specify the time units; that being said I'm not sure how keen we are in adding more arguments to that interface.  
- It also might be possible to do this automatically (by searching for one of `['hours', 'days']` in the units string of the original file).  I would lean towards this solution, with sufficient testing.

For the time being (since I don't switch back and forth between doing high-frequency and low-frequency analysis too often) it's fine for me to correct this by changing the hard-coding (e.g. change the time units to `"hours since 1900-01-01 00:00:00"`), but I definitely think this is something that needs to be addressed eventually (although hopefully at some point this gets fixed upstream, so we don't have to use a workaround to begin with).

**Comment by spencerahill** [comment]
Good catch.  How about something like this:

``` python
freq = test[v].attrs['units'].split('since')[0]
test[v].attrs['units'] = freq + 'since 1900-01-01 00:00:00'
```

Haven't tested this even informally but seems like it should work.

**Comment by spencerkclark** [comment]
:+1: That looks like a very clean way to go.  I'll give it a try tomorrow.

That's also more or less how it's done in xarray: https://github.com/pydata/xarray/blob/master/xarray/conventions.py#L83-L91

**Comment by spencerahill** [comment]
Nice.  I'm sure the way they do it is more robust, so you can emulate them directly if you'd like.  Your call if its worth the trouble -- as you noted, for us this is ultimately a temporary workaround anyways.

**Comment by spencerkclark** [comment]
@spencerahill by the way, your suggested workaround is good in my case -- the only possible advantage I see in using the xarray method is that it is robust to extraneous whitespace.  Beyond that, I think the two methods basically do the same thing.

**Comment by spencerahill** [comment]
:+1:

**Comment by spencerahill** [comment]
Closed in #73
